Tests/tests/test_OCIContainer.cpp

```diff
@@ -0,0 +1,220 @@
+#include "DobbyMock.h"
+#include "FactoriesImplementation.h"
+
+using namespace WPEFramework;
+using ::testing::NiceMock;
+
+namespace{
+    const int32_t descriptor1 = 91;
+    const int32_t descriptor2 = 94;
+    const string containerId1 = "com.bskyb.epgui";
+    const string containerId2 = "Netflix";
+    const int stateRunning = 2;
+    const int listenerId = 5;
+}
+
+class OCIContainerTest : public ::testing::Test {
+protected:
+    Core::ProxyType<Plugin::OCIContainer> plugin;
+    Core::JSONRPC::Handler& handler;
+    Core::JSONRPC::Connection connection;
+    string response;
+
+    OCIContainerTest()
+        : plugin(Core::ProxyType<Plugin::OCIContainer>::Create())
+        , handler(*(plugin))
+        , connection(1, 0)
+        {
+        }
+        virtual ~OCIContainerTest() = default;
+};
+
+class OCIContainerInitializedTest : public OCIContainerTest {
+protected:
+    NiceMock<ServiceMock> service;
+    DobbyProxyMock dobbymock;
+    IpcServiceMock ipcservicemock;
+
+    OCIContainerInitializedTest()
+        : OCIContainerTest()
+    {
+        DobbyProxy::getInstance().impl = &dobbymock;
+        IpcService::getInstance().impl = &ipcservicemock;
+
+        EXPECT_CALL(ipcservicemock, start())
+            .WillOnce(::testing::Return(true));
+
+        EXPECT_CALL(dobbymock, registerListener(::testing::_, ::testing::_))
+            .WillOnce(::testing::Return(listenerId));
+
+        EXPECT_EQ(string(""), plugin->Initialize(&service));
+    }
+
+    virtual ~OCIContainerInitializedTest() override
+    {
+        EXPECT_CALL(dobbymock, unregisterListener(listenerId))
+            .WillOnce(::testing::Return());
+
+        plugin->Deinitialize(&service);
+    }
+};
+
+
+
+TEST_F(OCIContainerTest, RegisteredMethods){
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("listContainers")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getContainerState")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getContainerInfo")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("startContainer")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("startContainerFromDobbySpec")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("stopContainer")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("pauseContainer")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("resumeContainer")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("executeCommand")));
+ }
+
+ TEST_F(OCIContainerInitializedTest, listContainersTest){
+
+    std::list<std::pair<int32_t, std::string>> containerslist = {{descriptor1, containerId1},{descriptor2, containerId2}};
+    EXPECT_CALL(dobbymock, listContainers())
+        .WillOnce(::testing::Return(containerslist));
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("listContainers"), _T("{}"), response));
+    EXPECT_EQ(response, string("{\"containers\":[{\"Descriptor\":91,\"Id\":\"com.bskyb.epgui\"},{\"Descriptor\":94,\"Id\":\"Netflix\"}],\"success\":true}"));
+}
```

Comment:
This doesn't seem like you understand the concern. When you set id 91 and check id 91 it is fine, reader understands it is the same id. When you set 2 and check "Running" it's not fine, reader doesn't understand 2 = "Running".  When you set "descriptor1" and check 91 it's not fine, reader doesn't understand "descriptor1" = 91. When you set "containerId2" and check "Netflix" it's not fine, reader doesn't  understand "containerId2" = "Netflix". I hope you can see the point.